### PR TITLE
refactor(unlock-app, locksmith): Changed data model for events

### PR DESCRIPTION
**Pull request by julien51**
# Description

We used to assume 1 event per lock and all of our event application was built around lock where the metadata for the event is stored in the lock object.
We now have an `event` object that itself points to locks (through a checkout config) and we can build a better UX.

This is a draft..

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by julien51** [comment]
This needs https://github.com/unlock-protocol/unlock/pull/12849 to be rebased.

File: unlock-app/src/hooks/useEventOrganizer.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            lockAddress,
+            account!,
+            checkoutConfig.locks[lockAddress].network
+          )
+        )
+      )
+      return !!isManagerByLock.find((isManager) => !!isManager)
+    }
+  )
+}
```
Comment (searchableguy) [inline unlock-app/src/hooks/useEventOrganizer.ts:37]: Don't we already have a similar hook?

File: unlock-app/src/pages/event/[slug].tsx [hunk 2]
```diff
@@ -10,47 +12,26 @@ interface Params {
...
-  const { lockAddress, network, metadata } = p.pageProps
-  return (
-    <EventContentWithProps
-      lockAddress={lockAddress}
-      network={network}
-      metadata={metadata}
-    ></EventContentWithProps>
-  )
+const EventPage = (props: EventPageProps) => {
+  console.log(props.pageProps)
```
Comment (searchableguy) [inline unlock-app/src/pages/event/[slug].tsx:33]: remove

File: unlock-app/src/services/ethpass.ts [hunk 3]
```diff
@@ -26,46 +26,6 @@ export const isEthPassSupported = (network: number) => {
   ].includes(network)
 }
 
-export const applePass = (
```
Comment (julien51) [inline unlock-app/src/services/ethpass.ts:29]: This is not used, so I am cleaning it up!

File: unlock-app/src/hooks/useEventOrganizer.ts (hunk 1 above)
Comment (julien51) [reply unlock-app/src/hooks/useEventOrganizer.ts]: Not exactly. We have something to check for a single lock. I was initially wondering if we could re-use, but I thought it was also more useful to have a different name/function for clarity.

File: unlock-app/src/pages/event/[slug].tsx (hunk 2 above)
Comment (julien51) [reply unlock-app/src/pages/event/[slug].tsx]: Done :)

File: unlock-app/src/components/content/EventContent.tsx [hunk 4]
```diff
@@ -9,6 +9,8 @@ import EventDetails from './event/EventDetails'
 import { EventLandingPage } from './event/EventLandingPage'
 import { useRouterQueryForLockAddressAndNetworks } from '~/hooks/useRouterQueryForLockAddressAndNetworks'
 import { useMetadata } from '~/hooks/metadata'
+import { toFormData } from '~/components/interface/locks/metadata/utils'
+import { Event, PaywallConfigType } from '@unlock-protocol/core'
 
 export const EventContent = () => {
   const {
```
Comment (julien51) [inline unlock-app/src/components/content/EventContent.tsx:16]: This is the legacy path (before slugs existed on locks...) We have ti reconstruct the whole object from the lock address.

**Pull request merged**